src/twisted/application/app.py

```diff
@@ -392,8 +395,14 @@ def startReactor(self, reactor, oldstdout, oldstderr):
 
         @see: L{runReactorWithLogging}
         """
+        if reactor is None:
+            from twisted.internet import reactor
         runReactorWithLogging(
             self.config, oldstdout, oldstderr, self.profiler, reactor)
+        try:
+            self._exitSignal = reactor._exitSignal
+        except AttributeError:
```

Comment:
Furthermore…

Given that `_exitSignal` is a private attribute, it's always possible that some 3rd-party reactor is using it.

As such, an unambiguous way to signal [namespace ownership](https://glyph.twistedmatrix.com/2017/05/the-sororicide-antipattern.html) would be helpful here.  Interfaces can do this.

Basically, you make a private interface, `_ISupportsExitSignaling`, make every reactor in Twisted that uses this functionality provide it, and then here you can do `if _ISupportsExitSignaling.providedBy(reactor)` before accessing this attribute.  This is unambiguous, and makes it easier for individual reactors to document what they're doing when they provide this attribute.  How does this sound?